<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Estimate genome size.pl by josephryan</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/pygment_trac.css">
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>Estimate genome size.pl</h1>
        <p>scripts to estimate genome size and coverage from kmer distribution generated by jellyfish</p>

        <p class="view"><a href="https://github.com/josephryan/estimate_genome_size.pl">View the Project on GitHub <small>josephryan/estimate_genome_size.pl</small></a></p>


        <ul>
          <li><a href="https://github.com/josephryan/estimate_genome_size.pl/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/josephryan/estimate_genome_size.pl/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/josephryan/estimate_genome_size.pl">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
        <h3>Welcome to estimate_genome_size.pl GitHub Page</h3>

<p>These scripts can be used to estimate genome size and coverage from fastq files containing sequencing reads</p>

<p>you will need JELLYFISH: http://www.cbcb.umd.edu/software/jellyfish/</p>

<p>For jellyplot.pl you will need GNUPLOT: http://www.gnuplot.info/</p>

<p>A typical session:</p>

<pre><code>$ # count k-mers (see jellyfish documentation for options)
gzip -dc reads1.fastq.gz reads2.fastq.gz | jellyfish count -m 31 -o fastq.counts -C -s 10000000000 -U 500 -t 30 /dev/fd/0 

# generate a histogram
jellyfish histo fastq.counts_0 > fastq.counts_0.histo

# generate a pdf graph of the histogram
jellyplot.pl fastq.counts_0.histo

# look at fastq.counts_0.histo.pdf and identify the approximate peak

# use find_valleys.pl to help pinpoint the actual peak
find_valleys.pl fastq.counts_0.histo

# estimate the size and coverage
estimate_genome_size.pl --kmer=31 --peak=42 --fastq=reads1.fastq.gz reads2.fastq.gz
</code></pre>

<p>NOTES about the typical session:</p>
<p>1. it is helpful to run with multiple kmer sizes to see if this has an effect (change -m in first jellyfish command)</p>
<p>2. -s should be adjusted to the size of your RAM in the first jellyfish command (see <a href="http://www.cbcb.umd.edu/software/jellyfish/jellyfish-manual-1.1.pdf">jellyfish manual</a>)</p>
<p>3. The first jellyfish command includes '-U 500' this tells jellyfish not to output k-mer with count > 500. It is recommended to run without this the first time and then rerun with an uppercount that includes the first peak. This makes the files much easier to manage.</p>

<h3>See the following for the principle behind the script</h3>
<p><a href="https://banana-slug.soe.ucsc.edu/bioinformatic_tools:jellyfish">https://banana-slug.soe.ucsc.edu/bioinformatic_tools:jellyfish</a></p>

<p><a href="http://seqanswers.com/forums/archive/index.php/t-10988.html">http://seqanswers.com/forums/archive/index.php/t-10988.html</a></p>

<p><a href="https://banana-slug.soe.ucsc.edu/bioinformatic_tools:quake">https://banana-slug.soe.ucsc.edu/bioinformatic_tools:quake</a></p>

<h3>If you don't believe this works</h3>

<p>Simulate some next generation data based on an already-sequenced genome. I used the program ART to do this with the Hydra magnipapillata genome:</p>

<p><a href="http://www.niehs.nih.gov/research/resources/software/biostatistics/art/">http://www.niehs.nih.gov/research/resources/software/biostatistics/art/</a></p>

<p><pre>art_illumina --paired --in Hm_genomic.fa --out Hm_genomic_wildtype --len 100 --fcov 43 --mflen 180 --sdev 3.5</pre></p>

<p>I also simulated perfect paired-end data with a sliding window. In both cases I recovered the correct coverage and size.</p>

<h3>Authors and Contributors</h3>

<p>Joseph F. Ryan, Ph.D. (<a href="https://github.com/josephryan" class="user-mention">@josephryan</a>)</p>

<h3>Support or Contact</h3>

<p>There is <a href="https://github.com/josephryan/estimate_genome_size.pl/wiki/Documentation">Documentation</a> on the wiki.</p>

<p>If you have a negative or positive experience with estimate_genome_size.pl, I am interested in hearing about it. E-Mail: josephryan@yahoo.com or leave something on the <a href="https://github.com/josephryan/estimate_genome_size.pl/wiki">wiki</a>.</p>

<h3>Citing estimate_genome_size.pl</h3>

<p>Ryan, J. F. (2013). estimate_genome_size.pl (version 0.03) [Computer software]. Bergen, Norway: Sars International Centre for Marine Molecular Biology. Retrieved from http://josephryan.github.com/estimate_genome_size.pl/
      </section>
      <footer>
      <footer>
        <p>This project is maintained by <a href="https://github.com/josephryan">josephryan</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
    
  </body>
</html>
